jscomp/test/variantsMatching.res

```diff
@@ -272,3 +272,17 @@ module CustomTagNotInline = {
   let a = A(10)
   let b = B(20)
 }
+
+module UntaggedWithBool = {
+  @unboxed @genType
+  type t = String(string) | Float(float) | Bool(bool) | Object({name: string})
```

Comment:
Unrelate: this defines `string` twice.
It should be an error.

Reply:
Where does it define string twice? The object prop `name` + the constructor..?

Reply:
I guess it doesn't look right with this output.
```ts
export type UntaggedWithBool_t = string | number | boolean | string;
```

Reply:
Ahh, the two comments are connected! Now I see.